src/index.js

```diff
@@ -135,6 +134,65 @@ class IpfsRepo {
     })
   }
 
+  /**
+   * Returns the repo locker to be used. Null will be returned if no locker is requested
+   *
+   * @private
+   * @returns {Locker}
+   */
+  _getLocker () {
+    if (this.options.locker === false) {
+      return null
+    } else if (this.options.locker) {
+      return this.options.locker
+    }
+
+    assert(lockers[this.options.lock], 'Unknown lock type: ' + this.options.lock)
+    return lockers[this.options.lock]
+  }
+
+  /**
+   * Creates a lock on the repo if a locker is specified. The lockfile object will
+   * be returned in the callback if one has been created.
+   *
+   * @param {string} path
+   * @param {function(Error, lockfile)} callback
+   * @returns {void}
+   */
+  _openLock (path, callback) {
+    if (this._locker) {
```

Comment:
also 👎 on being able to disable the locking, this is just asking for trouble

Reply:
I am fine with leaving locking required, the custom locker is the needed piece. It's fairly easy to subvert locking at the moment even if we don't explicitly allow it to be disabled. With custom datastore support there is a potential use case for users wanting to handle locking at the object level, rather than the repo level.

If we decide to leave it required, I can update the documentation when I add in the custom locker docs to encourage users to use locking appropriate for their use case and custom datastores, such as storing their lock on S3 if their datastore is there.